.clang-tidy

```diff
@@ -0,0 +1,13 @@
+Checks: '-*,bugprone-*,cert-*,clang-analyzer-*,cppcoreguidelines-*,hicpp-*,misc-*,modernize-*,readability-*'
```

Comment:
Let's reduce the number of checks. Could you review the logs to identify the most useful diagnostics and retain only those checks for now (perhaps just `cppcoreguidelines` and `bugprone`)? This will allow us to address the issues and then run clang-tidy with a minimal configuration on each PR.

Reply:
I disabled some checks that were noisy or not really useful. For others, fixes are provided. From what I saw, I think these two are useful `misc-include-cleaner`, `misc-unused-parameters`, but before enabling we need to fix their warnings.